src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -40,23 +40,29 @@ public void init() {
   }
 
   /**
-   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
-   * an empty array if the user is not provided.
+   * Responds with a JSON representation of {@link Message} data for logged-in user and another user. Responds with
+   * an empty array if the user is not logged in or another user is not provided.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     response.setContentType("application/json");
 
-    String user = request.getParameter("user");
-
-    if (user == null || user.equals("")) {
+    // Get logged-in user and other user
+    String loggedInUser = null;
+    String otherUser = request.getParameter("user");
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      loggedInUser = userService.getCurrentUser().toString();
+    }
+    
+    if (loggedInUser == null || loggedInUser.equals("") || otherUser == null || otherUser.equals("")) {
```

Comment:
For error cases, the common practice in web services is to set an HTTP status code that indicates an error.

There is a list of HTTP status codes here:
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

For the case where the user is not logged-in, the expected code would be:
401 ("Unauthenticated")

For the case where the required GET parameter is absent, the expected code would be:
400 ("Bad Request"), which is logically equivalent to "InvalidArgumentException"

You can set that status code with the `setStatus` function as in:

     if (!isValidUser(loggedInUser)) {
          response.setStatus(401);
          response.getWriter().println(...);
          return;
     }
     if (!isValidUser(otherUser)) {
          response.setStatus(400);
          response.getWriter().println(...);
          return;
     }